packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewerContainer.js

```diff
@@ -69,7 +69,6 @@ class SingleImageViewerContainer extends React.Component {
   async preload () {
     const { subject } = this.props
     if (subject && subject.locations) {
-      // TODO: Validate for allowed image media mime types
       const imageUrl = Object.values(subject.locations[0])[0]
```

Comment:
Where is this URL being validated?

Reply:
`createLocationCounts` now uses the validation and if any of the locations are invalid, an empty object is returned. There's a conditional in the getter in the `Subject` store that will now evaluate as false and the viewer will return null. Then  in the `SubjectViewer` component, the return JSX component will be null, so this code will never run. I can add a test in that component to be sure.

Reply:
I've added a couple of tests in `Subject.spec.js` for when the subject is invalid, the viewer should be null.